src/client/Microsoft.Identity.Client/Internal/JsonWebToken.cs

```diff
@@ -131,45 +187,6 @@ private string Encode(X509Certificate2 certificate, string base64EncodedThumbpri
             return string.Concat(encodedHeader, ".", encodedPayload);
         }
 
-        [JsonObject]
-        [Preserve(AllMembers = true)]
-        internal class JWTHeader
```

Comment:
This flow is too slow, as it relies on JSON serialization. But in fact the JSON is just a string.